### PR TITLE
use type checkers in `fit.R`

### LINKED ISSUE #1181: transition to rlang type checkers in `fit.R`

### PR DESCRIPTION
**Pull request by simonpcouch**
Closes #1181.🐝

### CODE REVIEW

File: R/misc.R [hunk 1]
```diff
@@ -533,6 +533,21 @@ check_case_weights <- function(x, spec, call = rlang::caller_env()) {
   invisible(NULL)
 }
 
+# ------------------------------------------------------------------------------
+
+check_inherits <- function(x, cls, arg = caller_arg(x), call = caller_env()) {
```
Comment (simonpcouch) [inline R/misc.R:538]: Decided to write out `inher()` in favor of this helper. I think we could probably get a good bit of mileage out of this based on the number of `if (inherits(` in this repo.

File: R/fit.R [hunk 2]
```diff
@@ -373,59 +370,47 @@ eval_mod <- function(e, capture = FALSE, catch = FALSE, envir = NULL, ...) {
...
-  invisible(x)
-}
-
-# ------------------------------------------------------------------------------
-
-check_interface <- function(formula, data, cl, model) {
-  inher(formula, "formula", cl)
-  inher(data, c("data.frame", "dgCMatrix", "tbl_spark"), cl)
+check_interface <- function(formula, data, cl, model, call = caller_env()) {
+  check_inherits(formula, "formula", call = call)
```
Comment (EmilHvitfeldt) [inline R/fit.R:374]: couldn't this be `check_formula()`?

File: R/misc.R (hunk 1 above)
Comment (EmilHvitfeldt) [reply R/misc.R:538]: 100%, and the name is much better too!

File: R/fit.R [hunk 3]
```diff
@@ -444,7 +429,7 @@ check_xy_interface <- function(x, y, cl, model) {
 
   check_outcome(y, model)
 
-  cli::cli_abort("Error when checking the interface")
+  cli::cli_abort("Error when checking the interface", call = call)
```
Comment (EmilHvitfeldt) [inline R/fit.R:432]:
```suggestion
  cli::cli_abort("Error when checking the interface.", call = call)
```

**Review by EmilHvitfeldt — APPROVED** [review]
so happy to see this being done!

File: R/fit.R (hunk 2 above)
Comment (simonpcouch) [reply R/fit.R:374]: omg totally

**Pull request merged**